funfact/lang/_tensor.py

```diff
@@ -17,16 +17,20 @@ def symbol(self):
     @symbol.setter
     def symbol(self, string: str):
         m = re.fullmatch(r'([a-zA-Z]+)(?:_(\d+))?', string)
-        if m is None:
-            raise RuntimeError(
-                f'{repr(string)} is not a valid symbol.\n'
-                'A symbol must be alphabetic and optionally followed by an '
-                'underscore and a numeric subscript. '
-                'Examples: i, j, k_0, lhs, etc.'
-            )
-        self._letter, self._number = m.groups()
-        if self._letter == 'Anonymous':
-            self._letter = r'\varphi'
+        try:
+            self._letter, self._number = m.groups()
+        except AttributeError:
+            m = re.fullmatch(r'__(\d+)', string)
```

Comment:
Why do you check for symbols with double underscores? Is this related to #4 ?

Reply:
Right. This is still a temporary solution. The plan is to allow truly anonymous indices, and then only assign readable letters to them when rendering in latex/ascii.